### ISSUE TITLE
issue parsing xlsx (formatting?)

### ISSUE THREAD

**Issue by crepon-philipp** [martijn/xsv#18]
Hello when reading the following file only the numeric values are getting parsed correctly. all text-values are parsed to nil
[formatting_issue.xlsx](https://github.com/martijn/xsv/files/5301838/formatting_issue.xlsx)

this is what I did:
`      x = Xsv::Workbook.open(file_path)`
`      sheet = x.sheets[0]`
`      # Iterate over rows`
`      sheet.each_row do |row|`
`        p '-----------------', row # => ["header1", "header2"], etc.`
`      end`

This is what is printed:
`"-----------------"`
`[nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil]`
`"-----------------"`
`[nil, nil, nil, nil, nil, 0, nil, nil, nil, nil, 1, nil, nil, nil, nil, nil, nil, nil, 1, nil, nil, nil, nil, nil, nil, nil, nil, 1, nil]`
`

when copying all data from the xlsx into a simple txt-editor and then copying it back into excel. the data is getting parsed correctly:
`"-----------------"`
`["PartNumber", "Brand", "SubBrand", "PopulationStatus", "PartType", "UniversalPart", "BasePartNumber", "Gtin", "GtinQualifierCode", "GtinQualifierName", "QuantitySize", "QuantitySizeUom", "ContainerType", "QuantityPerApplication", "QuantityPerApplicationUom", "QuantityPerApplicationQualifier", "EffectiveDate", "AvailableDate", "MinimumOrderQuantity", "MinimumOrderQuantityUom", "PartGroup", "PartSubGroup", "PartCategoryCode", "Unspsc", "VmrsCode", "Position", "DefaultMfrLabel", "DefaultApplicationQty", "Tags"]`
`"-----------------"`
`["P76550M-2", 1234, 1234, "Active (Publish to receiver database)", "Fuel Pump Module Assembly", 0, nil, 19826103970, "UP", "UPC", 1, "EA", "BX", 1, "EA", "MAX", nil, Fri, 23 Aug 2019, 1, "EA", nil, nil, 550303, 40151532, 44003258, nil, nil, 1, nil]
`

**Comment by crepon-philipp** [comment]
just noticed that just changing something in that file and saving it again is already enough in order to get the file parse correctly. But what could be wrong with that file then? in excel it is showing all correct?

**Comment by crepon-philipp** [comment]
FYI: I found the root cause why this issue came up for me. it was related to the way the file was created. as mentioned here:
https://github.com/Paxa/fast_excel/issues/63#issue-711535354

But I didn't have this issue with the previous xlsx-parser I used for importing. So I think this is something that could be fixed in this gem as well.
(For reference. the previous xlsx parser I used for reading excel-files was creek. But that gem has some pending issues that didn't get resolved why I had to switch:
[](https://github.com/pythonicrubyist/creek/issues)

**Comment by martijn** [comment]
```
      <c r="A1" t="inlineStr">
        <is>
          <t>PartNumber</t>
        </is>
      </c>
```

This uses an inline string format I haven't seen before, documented here:

https://docs.microsoft.com/en-us/dotnet/api/documentformat.openxml.spreadsheet.inlinestring?redirectedfrom=MSDN&view=openxml-2.8.1

There's some esoteric stuff in here, but I'll make an update soon to deal with this plain text at least.

**Comment by martijn** [comment]
@crepon-philipp could you test against master and see if this solves your issue?

**Comment by crepon-philipp** [comment]
@martijn Hmm, Looking still the same

**Comment by martijn** [comment]
Weird.. are you sure you are using the master branch from GitHub? I have not yet released a new version.

This is what I get:

```
irb(main):001:0> w = Xsv::Workbook.open("formatting_issue.xlsx")
irb(main):002:0> w.sheets[0][0]
=> ["PartNumber", "Brand", "SubBrand", "PopulationStatus", "PartType", "UniversalPart", "BasePartNumber", "Gtin", "GtinQualifierCode", "GtinQualifierName", "QuantitySize", "QuantitySizeUom", "ContainerType", "QuantityPerApplication", "QuantityPerApplicationUom", "QuantityPerApplicationQualifier", "EffectiveDate", "AvailableDate", "MinimumOrderQuantity", "MinimumOrderQuantityUom", "PartGroup", "PartSubGroup", "PartCategoryCode", "Unspsc", "VmrsCode", "Position", "DefaultMfrLabel", "DefaultApplicationQty", "Tags"]
irb(main):003:0> w.sheets[0][1]
=> ["P76550M-2", "1234", "1234", "Active (Publish to receiver database)", "Fuel Pump Module Assembly", 0, nil, "00019826103970", "UP", "UPC", 1, "EA", "BX", "1", "EA", "MAX", nil, "2019-08-23", 1, "EA", nil, nil, "550303", "40151532", "044003258", nil, nil, 1, nil]
```

**Comment by crepon-philipp** [comment]
My bad. Using master it is now working as expected 👍

**Comment by martijn** [comment]
Great. I completed the support and released Xsv 0.3.18 with these changes.

Thanks for using Xsv!
